operator/controllers/loki/lokistack_controller.go

```diff
@@ -224,3 +235,31 @@ func (r *LokiStackReconciler) enqueueAllLokiStacksHandler() handler.EventHandler
 		return requests
 	})
 }
+
+func (r *LokiStackReconciler) enqueueForUserWorkloadAMService() handler.EventHandler {
+	ctx := context.TODO()
+	return handler.EnqueueRequestsFromMapFunc(func(obj client.Object) []reconcile.Request {
+		lokiStacks := &lokiv1.LokiStackList{}
+		if err := r.Client.List(ctx, lokiStacks); err != nil {
+			r.Log.Error(err, "Error getting LokiStack resources in event handler")
+			return nil
+		}
+		var requests []reconcile.Request
+
+		if obj.GetName() == openshift.MonitoringSVCOperated && obj.GetNamespace() == openshift.MonitoringUserwWrkloadNS {
+
+			for _, stack := range lokiStacks.Items {
+				requests = append(requests, reconcile.Request{
+					NamespacedName: types.NamespacedName{
+						Namespace: stack.Namespace,
+						Name:      stack.Name,
+					},
+				})
+			}
```

Comment:
I suggest to omit any stack that has tenant mode other than `openshift-logging` or `openshift-network`. These are the only two tenants we reconcile OpenShift related Alertmanager configuration.

Reply:
done